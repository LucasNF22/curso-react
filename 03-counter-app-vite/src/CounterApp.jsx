import { useState } from 'react'
import PropTypes from 'prop-types';

export const CounterApp = ({ value }) => {

    const [ counter, setCounter ] = useState( value );

    const handleSuma = () => {
        // console.log(event);
        setCounter( counter +1 )
    }

    // Forma corta de escribir
    const handleResta = () => setCounter( counter -1 );
     
    const handleReset = () => setCounter( value );

    return (
        <>
            <h1>CounterApp</h1>
            <h1 aria-label="counter" > { counter } </h1>
            <button  onClick={ handleResta } >-1</button>
            <button onClick = { handleSuma } >+1</button>
            <button  id="reset" onClick={ handleReset }>reset</button>
        </>
    )
}

CounterApp.propTypes = {
    value: PropTypes.number.isRequired,
};