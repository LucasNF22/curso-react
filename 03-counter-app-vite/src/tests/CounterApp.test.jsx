import { fireEvent, getByRole, render, screen } from "@testing-library/react";
import { CounterApp } from "../CounterApp";


describe('Pruebas en CounterApp', () => {

    const value = 100;

    test('Debe hcaer match con el snapxhot', () => {

        const { container } = render( <CounterApp value={ value } /> )
        expect( container ).toMatchSnapshot();
    });

    test('Debe mostrar un valor inicial de 100', () => {
        
        render( <CounterApp value={ value } />);
        expect( screen.getByText(100) ).toBeTruthy();
        
    });

    test('Debe incrementar con el boton +1', () => {

        render( <CounterApp value={ value } />);
        fireEvent.click( screen.getByText('+1') );
        expect( screen.getByRole( 'heading' , { name: 'counter' }).innerHTML ).toContain("101")
    });

    test('Debe decrementar con el boton -1', () => {

        render( <CounterApp value={ value } />);
        fireEvent.click( screen.getByText('-1') );
        expect( screen.getByRole( 'heading' , { name: 'counter' }).innerHTML ).toContain("99")

    });
    
    test('Debe funcionar el boton de Reset', () => {

        render( <CounterApp value={ 50 } />);

        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        fireEvent.click( screen.getByText('+1') );
        
        fireEvent.click(screen.getByRole( 'button' , { name: 'reset' }));
        
        expect( screen.getByRole( 'heading' , { name: 'counter' }).innerHTML ).toContain("50")

    });

});