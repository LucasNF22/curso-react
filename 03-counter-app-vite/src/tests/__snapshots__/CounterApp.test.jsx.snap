// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en CounterApp Debe hcaer match con el snapxhot 1`] = `
<div>
  <h1>
    CounterApp
  </h1>
  <h1
    aria-label="counter"
  >
     
    100
     
  </h1>
  <button>
    -1
  </button>
  <button>
    +1
  </button>
  <button
    id="reset"
  >
    reset
  </button>
</div>
`;
